element.go

```diff
@@ -441,7 +454,7 @@ var endOfDataElement = &Element{Tag: dicomtag.Tag{Group: 0x7fff, Element: 0x7fff
 // element is a pixel data, or it sees an element defined by options.StopAtTag.
 //
 // - On successful parsing, it returns non-nil and non-endOfDataElement value.
-func ReadElement(d *dicomio.Decoder, options ReadOptions) *Element {
+func ReadElement(d *dicomio.Decoder, parsedData *DataSet, options ReadOptions) *Element {
```

Comment:
Parsing some attributes, namely `PixelData` requires information encoded in previously parsed attributes (for example: rows, cols, bits per sample, etc). For exploration purposes, this information was injected into `ReadElement` so that it can be used to parse `PixelData` into a relatively meaningful form (more than just bytes), but it may also be useful to keep `ReadElement` independent of previous elements and do a special second parsing pass on the `PixelData` element to put it into a well parsed form...decision will be made after additional consideration and taking stock of this library and any refactors that may be needed

Reply:
though I find it very appealing for `ReadElement`'s API to not rely on previous data (though one can always pass in `nil` here if not parsing `PixelData` elements), that change may have to accompany other refactors that should probably happen in future diffs. #11

Reply:
switch order of `parsedData *DataSet, options ReadOptions` so that `parsedData=nil` can be the last argument if/when called with `nil`